app/models/financialDetails/FinancialDetailsResponseModel.scala

```diff
@@ -77,6 +78,20 @@ case class FinancialDetailsModel(balanceDetails: BalanceDetails,
       }
   }
 
+  def isValidPenalty(documentId: String, penaltiesEnabled: Boolean): Boolean = {
```

Comment:
no changes in FinancialDetailsResponseModel, we are using ChargeItem or/and any sub-type of TransactionItem instead;
we would need to apply conversion to ChargeItem within HomeController->getDueDates method and extend ChargeItem / TransactionItem types where needed ;